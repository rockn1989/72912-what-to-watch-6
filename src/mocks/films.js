export default [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 2,
    name: `Aviator`,
    posterImage: `img/aviator.jpg`,
    previewImage: `img/aviator.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 2.9,
    scoresCount: 20,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Advanture`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 3,
    name: `Revenant`,
    posterImage: `img/revenant.jpg`,
    previewImage: `img/revenant.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 6.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 4,
    name: `Johnny English`,
    posterImage: `img/johnny-english.jpg`,
    previewImage: `img/johnny-english.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 5,
    name: `What we do is the shadow`,
    posterImage: `img/what-we-do-in-the-shadows.jpg`,
    previewImage: `img/what-we-do-in-the-shadows.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 3.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Comedy`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 6,
    name: `We need to talk about Kevin`,
    posterImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewImage: `img/we-need-to-talk-about-kevins.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2009,
    isFavorite: false
  },
  {
    id: 7,
    name: `Macbeth`,
    posterImage: `img/macbeth.jpg`,
    previewImage: `img/macbeth.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 8,
    name: `Bohemain Rhaapsody`,
    posterImage: `img/bohemian-rhapsody.jpg`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 9.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `History`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 9,
    name: `We need to talk about Kevin`,
    posterImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewImage: `img/we-need-to-talk-about-kevins.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2009,
    isFavorite: false
  },
  {
    id: 10,
    name: `Macbeth`,
    posterImage: `img/macbeth.jpg`,
    previewImage: `img/macbeth.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `Drama`,
    realeased: 2014,
    isFavorite: false
  },
  {
    id: 11,
    name: `Bohemain Rhaapsody`,
    posterImage: `img/bohemian-rhapsody.jpg`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    previewVideolink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 `,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 9.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: 99,
    genre: `History`,
    realeased: 2014,
    isFavorite: false
  },
];
