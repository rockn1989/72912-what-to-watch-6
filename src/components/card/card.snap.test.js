import React from 'react';
import {render, screen} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import Card from './card';

import films from '../../mocks/films';
const film = films[0];

describe(`Card`, () => {
  beforeAll(() => {
    window.HTMLMediaElement.prototype.play = () => {};
    window.HTMLMediaElement.prototype.load = () => {};
    window.HTMLMediaElement.prototype.pause = () => {};
  });

  it(`Should render card correctly`, () => {
    const history = createMemoryHistory();

    const {container} = render(
        <Router history={history}>
          <Card
            id={film.id}
            title={film.name}
            previewVideolink={film.videoLink}
            img={film.posterImage}
            onHoverHandler={jest.fn()}
            isActive={false}/>
        </Router>
    );

    expect(screen.getByText(film.name)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it(`Should render card correctly if card is active`, () => {
    const history = createMemoryHistory();

    const {container} = render(
        <Router history={history}>
          <Card
            id={film.id}
            title={film.name}
            previewVideolink={film.videoLink}
            img={film.posterImage}
            onHoverHandler={jest.fn()}
            isActive={true}/>
        </Router>
    );

    expect(screen.getByText(film.name)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
