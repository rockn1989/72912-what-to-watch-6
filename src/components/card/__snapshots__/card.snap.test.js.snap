// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render card correctly 1`] = `
<div>
  <article
    class="small-movie-card catalog__movies-card"
    id="1"
  >
    <a
      class="small-movie-card__link"
      href="/films/1"
    >
      <div
        class="small-movie-card__image"
      >
        <img
          height="175"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="280"
        />
      </div>
      <h3
        class="small-movie-card__title"
      >
        The Grand Budapest Hotel
      </h3>
    </a>
  </article>
</div>
`;

exports[`Card Should render card correctly if card is active 1`] = `
<div>
  <article
    class="small-movie-card catalog__movies-card"
    id="1"
  >
    <a
      class="small-movie-card__link"
      href="/films/1"
    >
      <div
        class="small-movie-card__image"
      >
        <video
          poster="img/the-grand-budapest-hotel-poster.jpg"
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4 "
          style="width: 280px; height: 175px; object-fit: cover;"
        />
      </div>
      <h3
        class="small-movie-card__title"
      >
        The Grand Budapest Hotel
      </h3>
    </a>
  </article>
</div>
`;
